README.md

```diff
@@ -1,38 +1,77 @@
 
 
 ![alt text](figures/figure-2-color.jpeg)
 
-## Dependencies
-- [Python 3.6](https://www.anaconda.com/download/#macos)
-- [PyTorch](https://pytorch.org/)
-- [PIL](https://pillow.readthedocs.io/en/5.3.x/)
+## Changes in this Release
+- Use Python [multiprocessing](https://docs.python.org/3.6/library/multiprocessing.html)
+to speed up patch creation.
+- Replace [os](https://docs.python.org/3.6/library/os.html) with
+[pathlib](https://docs.python.org/3.6/library/pathlib.html) module.
+- Added [argparse](https://docs.python.org/3.6/library/argparse.html) support.
+- Rewrote patch processing code using vector operations.
+- Fully documented all functions and methods with docstrings and type hints.
+- Compute image means and standard deviations for normalization online.
+- Increased efficiency of code by rewriting inefficient code.
+
+## Known Issues and Limitations
+- Patch creation code doesn't find any patches in images that are slightly larger than the specified patch size.
+- Only 1 GPU supported.
+- Should work, but not tested on Windows.
+- In cases where no crops are found for an image, empty directories are created. Current workaround uses `try` and `except` statements to catch errors.
+- Image reading code expects colors to be in the RGB space. Current workaround is to keep first 3 channels.
+
+## Requirements
+- [imageio](https://pypi.org/project/imageio/)
+- [NumPy 1.16](https://numpy.org/)
 - [OpenCV](https://opencv.org/)
-- [Scikit-Learn](https://scikit-learn.org/stable/install.html)
-- [Scipy](https://www.scipy.org/)
-- GPU
+- [OpenSlide](https://openslide.org/)
+- [OpenSlide Python](https://openslide.org/api/python/)
+- [pandas](https://pandas.pydata.org/)
+- [PIL](https://pillow.readthedocs.io/en/5.3.x/)
+- [Python 3.6](https://www.python.org/downloads/release/python-360/)
+- [PyTorch](https://pytorch.org/)
+- [scikit-image](https://scikit-image.org/)
+- [scikit-learn](https://scikit-learn.org/stable/install.html)
+- [SciPy](https://www.scipy.org/)
+- [NVIDIA GPU](https://www.nvidia.com/en-us/)
+- [Ubuntu](https://ubuntu.com/)
 
 # Usage
 
 Take a look at `code/config.py` before you begin to get a feel for what parameters can be changed.
 
+## Note on Expected WSI Input
+We expect the WSI input to be at the crop or tissue level, rather than the entire WSI. While the code will still work if entire WSIs are provided, the performance will likely be bad.
```

Comment:
I still don't get this. I used WSI for 3 of my projects, and they work. Why should the performance be bad on them?

Reply:
@jasonwei20 The idea of this comment is that you need annotations instead of labels for the entire slide.  When running this network on WSIs that only have slide-level labels, the performance was terrible since the majority class was normal.

Reply:
@JosephDiPalma I see your point but to be honest I don't like this sentence because 1. not sure if you can generalize that and 2. it's the first item users see in this readme, specifically "the performance will likely be bad" giving negative impression. Could you rephrase in non-negative way, and I think this should be better placed under Limitation section.